### PR TITLE
Wallet Extension Integration

### PR DESCRIPTION
**Pull request by h0ngcha0**
- Picked Muchen's work for extension wallet integration
- Fixed some UI issues by updating the extension wallet

### CODE REVIEW

File: bridge_ui/package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@alephium/walletconnect-provider": "^0.2.0-rc.1",
     "@alephium/web3": "^0.2.0-rc.7",
     "@certusone/wormhole-sdk": "file:../sdk/js",
+    "@h0ngcha0/get-alephium": "^0.0.9",
```
Comment (h0ngcha0) [inline bridge_ui/package.json:9]: I will try to setup the CI to publish on the @alephium domain

**Comment by Lbqds** [comment]
The test failed because of this: https://github.com/bufbuild/buf/issues/1380, we need to update the `buf` to the latest version.

**Comment by h0ngcha0** [comment]
> Member

Thanks for the fix! <3

File: bridge_ui/src/utils/alephium.ts [hunk 2]
```diff
@@ -1,143 +1,142 @@
...
+  }
 
-        const localTokenPoolId = await getLocalTokenPoolId(provider, tokenId)
-        return localTokenPoolId ? new TokenInfo(0, 'token', 'token') : undefined
-    } catch (error) {
-        console.log("failed to get alephium token info, error: " + error)
-        return undefined
+  const tokenAddress = toAlphContractAddress(tokenId)
+  try {
+    const group = await provider.addresses.getAddressesAddressGroup(tokenAddress)
```
Comment (polarker) [inline bridge_ui/src/utils/alephium.ts:121]: There should be an utility function for this in `web3`.

File: bridge_ui/src/utils/alephium.ts [hunk 3]
```diff
@@ -1,143 +1,142 @@
...
+      const decimals = parseInt((state.fields[6] as node.ValU256).value)
+      return new TokenInfo(decimals, symbol, name)
     }
+
+    const localTokenPoolId = await getLocalTokenPoolId(provider, tokenId)
+    return localTokenPoolId ? new TokenInfo(0, 'token', 'token') : undefined
+  } catch (error) {
+    console.log("failed to get alephium token info, error: " + error)
+    return undefined
+  }
```
Comment (polarker) [inline bridge_ui/src/utils/alephium.ts:135]: This function is for testing purpose only right?

later we will read such information from a public repo `token-list`.

**Review by polarker — APPROVED** [review]
LGTM

File: bridge_ui/src/utils/alephium.ts (hunk 3 above)
Comment (Lbqds) [reply bridge_ui/src/utils/alephium.ts:135]: Yes, it's only for testing now, wormhole also has a repo for this: https://github.com/certusone/wormhole-networks.

**Pull request merged**